### PR TITLE
Flutter 3.7.0

### PR DESCRIPTION
**Pull request by domesticmouse**
Fixes: https://github.com/flutter/samples/issues/1443

## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Contributors Guide]: https://github.com/flutter/samples/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by domesticmouse** [comment]
FYI @gaaclarke @Renzo-Olivares I'm rolling all of the `beta` changes into a single PR to make it easier to land when `stable` increments. Adding y'all as reviewers so you can check that I haven't goofed your changes.

File: simplistic_editor/lib/basic_text_input_client.dart [hunk 1]
```diff
@@ -785,6 +791,12 @@ class BasicTextInputClientState extends State<BasicTextInputClient>
       ),
     );
   }
+
+  @override
+  void didChangeInputControl(
+      TextInputControl? oldControl, TextInputControl? newControl) {
+    // TODO(Renzo-Olivares): implement didChangeInputControl
```
Comment (domesticmouse) [inline simplistic_editor/lib/basic_text_input_client.dart:798]: PTAL @Renzo-Olivares

File: tool/flutter_ci_script_beta.sh [hunk 2]
```diff
@@ -15,21 +15,18 @@ declare -ar PROJECT_NAMES=(
...
+    "background_isolate_channels"
     "code_sharing/client"
     "code_sharing/server"
     "code_sharing/shared"
     "desktop_photo_search/fluent_ui"
     "desktop_photo_search/material"
-    # TODO(justinmc): To be enabled when its PR hits the master channel.
-    # https://github.com/flutter/flutter/pull/107193
-    # "experimental/context_menus"
+    "experimental/context_menus"
```
Comment (justinmc) [inline tool/flutter_ci_script_beta.sh:24]: Thanks for getting this for me 👍

File: simplistic_editor/lib/basic_text_field.dart [hunk 3]
```diff
@@ -104,8 +104,7 @@ class _BasicTextFieldState extends State<BasicTextField> {
         break;
     }
 
-    return FocusTrapArea(
-      focusNode: widget.focusNode,
+    return TextFieldTapRegion(
```
Comment (justinmc) [inline simplistic_editor/lib/basic_text_field.dart:107]: 👍

**Review by justinmc — APPROVED** [review]
LGTM for the text editing changes 👍.  @Renzo-Olivares might want to take a look at the simplistic_editor stuff.

**Review by Renzo-Olivares — APPROVED** [review]
LGTM

**Comment by domesticmouse** [comment]
PTAL: @QuncCccccc and @darrenaustin I've copied the contents of `experimental/material_3_demo` across to the top level with some minor edits.

**Review by craiglabenz — APPROVED** [review]

**Pull request merged**